javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -114,6 +114,25 @@ final class Formatter(private val dest: Appendable) extends Closeable with Flush
             result
           }
 
+          def formatThousands(value: String, decimalSep: Char = '.', thousandsGrouping: String = ","): String = {
+            val parts = value.split(decimalSep).toList
+            val integer = parts.head
+            val decimal = parts.tail
+            val headMod = integer.length % 3
+            val (headStr, tailStr) =
+              if (integer.length > 3 && headMod > 0)
+                (Some(integer.take(headMod)), integer.drop(headMod))
+              else (None, integer)
```

Comment:
There is no need to special case the empty head string. Just do:

```
val (headStr, tailStr) = integer.splitAt(headMod)
```

Also, consider reanming `headMod` to `headLen`.